Migrate to Mysql

gerli opened the pull request:
- The schema has been compacted and converted to use MySQL
- A base system for setting up Java servlets to replace the PL/SQL procedures has been added

gerli committed:
Convert database schema from Oracle to MySQL

* condense definitions
* remove prompts
* delete packages for the time being as they are not supported by MySQL
* change database name to CHARM
* convert varchar2 -> varchar, number -> int or float
* convert comments on stocking records into code comments

gerli committed:
Merge branch 'master' into gerli/migrate-to-mysql

gerli committed:
Revert column name to power

* had changed it to powera since editor seemed to suggest it was a reserved word (but seems to work with power)

gerli committed:
Add frost walker column to prices table

gerli committed:
Convert price and stock data to MySQL

* add USE statement
* remove prompts, alters and commits
* convert timestamps

gerli committed:
Add auto_increment to item_tree and stocking_records

* update current data to match auto_increment ids
* also reorder to run in order (otherwise won't match ids)

gerli committed:
Add automatic timestamping to date in stock records

gerli committed:
Merge branch 'master' into gerli/migrate-to-mysql

Conflicts:
	price_data.sql

MySQLified the enchant_info table as well

gerli committed:
Add a test JSP file for the database connection

* also include necessary jar files for the database connection to function

gerli committed:
Add a pure java servlet example, add web.xml and context.xml files

* note that adding a resource in context.xml also required dropping the mysql connector file in tomcat's lib directory
* no longer hardcoding database link in jsp
* declaring servlets in web.xml

gerli committed:
Add a java servlet database example

* this is basically the same as the .jsp, which allows to compare the two methods

gerli committed:
Add a simple charm database servlet

* currently takes the abbreviation of an enchantment and returns its full name

gerli committed:
Turn Charm servlet into an item ID checker

* currently has very rudimentary error handling
* doesn't normalise the ID, just checks that the individual ids are valid

gerli committed:
Add a simple build/deploy script

* compiles all the java files
* deploys (removing the old version) the app
* restarts tomcat

gerli committed:
Remove comment

ThatRedheadMC commented:
Waiting to merge until fully tested and there is a working example Charm.java :)

JamesGallicchio committed:
Add git pull to build.sh

- No need for manual git pull before bashing build :)

JamesGallicchio committed:
IMPORTANT: Use abbreviations in item_tree name column

* Replace full item and enchant names with 3/2 letter abbreviations, respectively
* Change name column to abbr, and limit to 3 character varchars
@gerli As in name, important. :P Changes schema, beware!

JamesGallicchio committed:
IMPORTANT: Use abbreviations in item_tree name column

* Replaces full item and enchant names with 3/2 letter abbreviations, respectively
* Changes name column to abbr, and limit to 3 character varchars
@gerli As in name, important. :P Changes schema, beware!

JamesGallicchio committed:
Merged branch gerli/migrate-to-mysql into gerli/migrate-to-mysql

JamesGallicchio committed:
Finish item_id_check, close #8

* Finishes conversion from SQL to Java
* Untested, needs testing but waiting on #7
* Adds functionality of returning ordered item ID

JamesGallicchio committed:
Fix string equality check

* Replaces == with string.equals()

gerli committed:
Add servlet-api.jar to lib directory

Adjust build to use that version. For some reason it is not present in tomcat's lib on the AWS machine

gerli committed:
Merged branch gerli/migrate-to-mysql into gerli/migrate-to-mysql

gerli committed:
Add missing imports and other fixes to make file compile

The Charm servlet doesn't seem to work but at least it compiles again now

gerli committed:
Switch to apache.commons.dbcp library

It seems that the Tomcat dbcp library is not supported on AWS https://forums.aws.amazon.com/thread.jspa?messageID=394470

gerli commented:
And now works on the server as well!

gerli committed:
Add primary key to ITEM_PRICES table

JamesGallicchio committed:
Charm.java itemidchecker fix

* Now increments iterable ResultSet (line 121) properly
* Should fix stacktrace in calls to itemidchecker :)

JamesGallicchio committed:
Charm.java itemidchecker fix

* Now increments iterable ResultSet (line 120) properly
* Should fix stacktrace in calls to itemidchecker :)

JamesGallicchio committed:
Merged branch gerli/migrate-to-mysql into gerli/migrate-to-mysql

JamesGallicchio committed:
Basic finalization of idinfochecker, closes #8

* Adds return statement (kinda important! :P)
* Fixes foundItem check to be 1 instead of 0
* Adds exception when something funky happens while checking ITEM_TREE paths which should never happen, but the compiler demands what the compiler wants
* Also includes recompiled classes

JamesGallicchio committed:
ITEM_TREE ID mishap fix

* Fixes ITEM_TREE IDs to be correct- they all seem to have been 1 too high, which was messing up the tree. Now fixed.
* Changes parentid checker in itemidchecker back to using 0 as root ID rather than 1
* Removes auto-increment from item_tree setup. There's a reason it wasn't auto incremented... if we are gonna change this info it's gonna be carefully rewritten by hand not autogenerated. The tree is completely dependent on each item having a set ID rather than any which ID that is auto-incremented.

JamesGallicchio committed:
Fix item parents in item_tree

* Changes parents from 1 to 0 for items in item_tree

JamesGallicchio committed:
Fix item/root logic in itemidchecker

* Changes root ID back to 0 rather than 1 :)
